src/main/java/com/atlassian/bitbucket/jenkins/internal/client/exception/RateLimitedException.java

```diff
@@ -0,0 +1,36 @@
+package com.atlassian.bitbucket.jenkins.internal.client.exception;
+
+import okhttp3.Headers;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The request was not accepted on the remote server due to rate limiting.
+ *
+ * @since 3.1
+ */
+public class RateLimitedException extends BitbucketClientException {
+
+    private long retryInMs;
+
+    public RateLimitedException(String message, int responseCode, @Nullable String body, Headers headers) {
+        super(message, responseCode, body);
+        String retryAfter = headers.get("Retry-After");
+        retryInMs = TimeUnit.SECONDS.toMillis(5);
+        if (retryAfter != null) {
+            try {
+                retryInMs = Long.parseLong(retryAfter) * 1000;
```

Comment:
Will Bitbucket's `Retry-After` header always be a number and never a date?

Reply:
yes, it is hard coded to 5

Reply:
I also went and looked at the [RFC](https://datatracker.ietf.org/doc/html/rfc6585#section-4), and it states:
>The response representations SHOULD include details explaining the
   condition, and MAY include a Retry-After header indicating how long
   to wait before making a new request.